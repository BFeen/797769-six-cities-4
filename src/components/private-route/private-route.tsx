import * as React from "react";
import {connect} from "react-redux";
import {Route, Redirect} from "react-router-dom";
import {getAuthorizationStatus} from "../../reducer/user/selectors";
import {AuthorizationStatus} from "../../reducer/user/user";
import {AppRoute} from "../../common/const";


interface Props {
  path: string;
  exact: boolean;
  authorizationStatus: string;
  render: () => React.ReactNode;
};

const PrivateRoute: React.FunctionComponent<Props> = (props) => {
  const {render, path, exact, authorizationStatus} = props;

  return (
    <Route
      path={path}
      exact={exact}
      render={() => {
        return (
          authorizationStatus === AuthorizationStatus.AUTH
            ? render()
            : <Redirect to={AppRoute.LOGIN} />
        );
      }}
    />
  );
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state)
});

export {PrivateRoute};
export default connect(mapStateToProps)(PrivateRoute);
