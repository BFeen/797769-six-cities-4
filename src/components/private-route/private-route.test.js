import React from "react";
import renderer from "react-test-renderer";
import {Router} from "react-router-dom";
import {PrivateRoute} from "./private-route.jsx";
import history from "../../history.js";


describe(`PrivateRoute snapshot test`, () => {
  it(`PrivateRoute rendering correctly`, () => {
    const tree = renderer.create(
        <Router history={history}>
          <PrivateRoute
            exact={true}
            path={`/favorite`}
            authorizationStatus={`AUTH`}
            render={() => {}}
          />
        </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`PrivateRoute redirected unauthorized user and rendering correctly`, () => {
    const tree = renderer.create(
        <Router history={history}>
          <PrivateRoute
            exact={true}
            path={`/favorite`}
            authorizationStatus={`NO_AUTH`}
            render={() => {}}
          />
        </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
