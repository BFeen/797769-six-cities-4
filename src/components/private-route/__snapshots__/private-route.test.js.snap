// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute snapshot test PrivateRoute redirected unauthorized user and rendering correctly 1`] = `null`;

exports[`PrivateRoute snapshot test PrivateRoute rendering correctly 1`] = `null`;
